src/messages/MNVMeofGwBeacon.h

```diff
@@ -92,10 +101,14 @@ class MNVMeofGwBeacon final : public PaxosServiceMessage {
     encode(gw_id, payload);
     encode(gw_pool, payload);
     encode(gw_group, payload);
-    encode(subsystems, payload);
+    encode(subsystems, payload, features);
     encode((uint32_t)availability, payload);
     encode(last_osd_epoch, payload);
     encode(last_gwmap_epoch, payload);
+    // Only encode sequence for enhanced beacons (version >= HEAD_VERSION)
+    if (get_header().version >= HEAD_VERSION) {
```

Comment:
First, this should be >= ENHANCED.  If you bump the version again, HEAD_VERSION will no longer be the correct check.

Second, COMPAT_VERSION is always set to LEGACY.  By doing that, you are stating that clients at LEGACY can skip this field no problem.  The testing you mention in https://github.com/ceph/ceph/pull/65563#discussion_r2454003001 suggests that you are both a) already sending ENHANCED encodings to old clients and that b) the clients deal with them fine (which is expected -- the encoding header includes a length so that old decoders can simply skip the fields they don't recognize).  I only expect to see conditional encodings like this when the client can't actually deal with them.  Why is this necessary?

Reply:
Do change HEAD_VERSION to ENCHANCED if you choose to keep this.